lib/rubocop/cop/rails/three_state_boolean_column.rb

```diff
@@ -46,6 +46,9 @@ def on_send(node)
 
             def_node = node.each_ancestor(:def, :defs).first
             table_node = table_node(node)
+
+            return unless table_node && column_node
```

Comment:
The following line calling `change_column_null?`, I read as: "if we can find code that changes the column nullness later, then don't add a violation". But if that code can't be found, because there's no enclosing `def` block to search, then the exception can't be made, and it will report a violation, maybe causing a false positive.

To be consistent with the above, if it can't find the nullness change because other parts of the context are missing (like the table name), I think it should also skip the exception and report a violation.

Reply:
Not 100% clear here, are you suggesting:

```return if def_node && change_column_null?(def_node, table_node&.value, column_node&.value)```

Or something else?

Reply:
I think I would go with

```ruby
return if def_node && table_node && column_node && \
  change_column_null?(def_node, table_node.value, column_node.value)
```